src/webgpu/util/device_pool.ts

```diff
@@ -297,6 +297,9 @@ class DeviceHolder implements DeviceProvider {
     let gpuValidationError: GPUValidationError | GPUOutOfMemoryError | null;
     let gpuOutOfMemoryError: GPUValidationError | GPUOutOfMemoryError | null;
 
+    // Submit to the queue to attempt to force a GPU flush.
+    this.device.queue.submit([]);
```

Comment:
Is this likely to be more effective than just awaiting popErrorScope? In principle that can't resolve until at least validation has finished.

Reply:
I think so - at least in Dawn, PopErrorScope doesn't look at the GPU queue or wait for anything at all - since all validation occurs on the "main" thread on the device timeline.

however, this could change if popErrorScope ever changes to wait for ALL enclosed operations (I remember there was some discussion about this for pipeline compilation and other stuff)

Reply:
got it, makes sense.